fiftyone/core/utils.py

```diff
@@ -2873,19 +2873,43 @@ def recommend_thread_pool_workers(num_workers=None):
     return num_workers
 
 
-def recommend_process_pool_workers(num_workers=None):
+def recommend_process_pool_workers(num_workers=None, default_num_workers=None):
     """Recommends a number of workers for a process pool.
 
-    If a ``fo.config.max_process_pool_workers`` is set, this limit is applied.
+    If this process is a daemon, the number of workers is 0.
+
+    If ``num_workers`` is None, the following order is used to determine the
+    number of workers:
+
+    - The configured (``fo.config``) default number of workers
+    - The passed-in default number of workers (``default_num_workers``)
+    - The system CPU count
+
+    If ``fo.config.max_process_pool_workers`` is set, this limit is applied.
 
     Args:
         num_workers (None): a suggested number of workers
+        default_num_workers (None): a default number of workers to use if
+            ``num_workers`` is None and no configured default is set
 
     Returns:
         a number of workers
     """
-    if num_workers is None:
-        num_workers = multiprocessing.cpu_count()
+    try:
+        if multiprocessing.current_process().daemon:
+            num_workers = 0
```

Comment:
This is confusing. If it needs to be 1 then why not just return 1

Reply:
Not entirely sure tbh I got this from the map samples stuff, I would have to look and see if `num_workers==1` makes sense or if 0 is required for some reason

Reply:
will fix to make more sense